### PR TITLE
fix issue with using EKSA_REGISTRY_MIRROR_CA env var

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Issue #, if available:*

*Description of changes:*
Environment variable `EKSA_REGISTRY_MIRROR_CA` was not being used to validate the cert. This was causing the CLI to fail validations in case of self-signed certs. This PR fixes that issue

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/639#" title="Author self-approved">abhinavmpandey08</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/api/v1alpha1/cluster.go
Comment (g-gaston) [inline pkg/api/v1alpha1/cluster.go:414]: could we add a test for this?

**Review by g-gaston — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/639#" title="Author self-approved">abhinavmpandey08</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/639#pullrequestreview-807699762" title="Approved">g-gaston</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08,g-gaston]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by vivek-koppuru — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/639#" title="Author self-approved">abhinavmpandey08</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/639#pullrequestreview-807699762" title="Approved">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/639#pullrequestreview-807905517" title="Approved">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08,g-gaston,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhinavmpandey08** [comment]
The unit test certificate was generated using the following commands and is valid only for `unit-test.local`
```
openssl genrsa -out ca.key 2048
openssl req -new -x509 -days 3650 -key ca.key -out ca.crt
openssl req -newkey rsa:2048 -nodes -keyout server.key -out server.csr
openssl x509 -req -extfile <(printf "subjectAltName=DNS:unit-test.local") -days 365 -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt
```

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/639#" title="Author self-approved">abhinavmpandey08</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/639#pullrequestreview-807699762" title="Approved">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/639#pullrequestreview-807984328" title="LGTM">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08,g-gaston,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
